src/System.Linq.Expressions/src/System/Runtime/CompilerServices/ReadOnlyCollectionBuilder.cs

```diff
@@ -23,7 +25,7 @@ public sealed class ReadOnlyCollectionBuilder<T> : IList<T>, System.Collections.
         private int _version;
 
         [NonSerialized]
-        private Object _syncRoot;
+        private object _syncRoot;
```

Comment:
As I understand SyncRoot is used in cases where one collection wraps another and both can be publicly available - then they need to expose the same locking object. Otherwise returning a priviate instance adds no value compared to just returning "this"